### PR TITLE
rfp proposal sbmission - JRPC Web Console

### PR DESCRIPTION
**Pull request by jrynkiew**
Submitting proposal for IoTeX rfp halogrant

### CODE REVIEW

**Comment by jam75010** [comment]
I cannot open my wallet the iOS application doesn't work ? What's wrong ?what I can do ?

**Comment by jrynkiew** [comment]
No, the wallet integration does not yet work. For now interface is complete n 30%

**Comment by jrynkiew** [comment]
> I cannot open my wallet the iOS application doesn't work ? What's wrong ?what I can do ?

Also, please be advised that recommended browser to use JRPC.pl with is Chrome / Mozilla Firefox / Edge / Opera.
I have not managed to integrate Apple Safari browser support yet, due to Safari Web Assembly support issues

**Comment by raullenchai** [comment]
Thanks for your interest in building on IoTeX, super excited! 

Can you elaborate more about what this JRPC.pl game looks like?

**Comment by jrynkiew** [comment]
Hi,

The game is divided into two main parts.

1. Hacking console + logs and ioPay integration. Users will be able to commit transactions to the IoTeX network via ioPay wallet, on the testnet as well as the mainnet. This console is designed to reward users for "hacking" activity (in layman's terms of course, as the hacking aspect will be scripted and non-linear, however bug bounty of actually finding bugs or hacks to bring down the network or the website itself will be rewarded as well).
This aspect of the console can be viewed as a game, with JRPC tokens as reward (mimo and other exchange integration forthcoming), however the plan is to integrate full iotex command line so that advanced users can actually integrate and perform fully legitimate complex operations via this console.
This aspect of the user experience is designed to bring and entice users to introduce them to advanced computing and computer science, as currently the vast majority of users are not aware of advanced features of blockchain networks, and rely 100% on user interfaces. Users will be allowed to interact with the live testnet without deploying their own iotex server, which is beyond the reach or technical acumen of most iotex users.

This is targeted at all ages, from children to introduce them to hacking (scripted scenarios, rewarded with JRPC tokens), as well as advanced users to actually utilize the network via web console.

2.A 2.5D platformer game in the background. This website is built on a GLew SDL2 Emscripten WebAssembly cross-compiled code from C/C++. In addition, it utilizes the imGui window library for rendering windows, icons and buttons. (ImGui will be responsible for rendering the "hacking" aspect of the game). In the background, SDL2 will be responsible for rendering game meshes and wireframes in conjunction with some low level GLew code. This platformer game is envisioned to run continuously in the background, even as the user continues his hacking console commands. The platformer is envisioned to be a 2D/2.5D Mario/Braid type game. Commands entered into the console can in the future trigger in-game events and even some real-world user interaction if the user opts-in via providing his username and password and user data such as e-mail and cell phone (not required). User can always close the console to play platformer. Each option is only a click away, accessible via menu or icon press. The platformer will reward user as well for progress via JRPC Hacking token, which then can be redeemed on mimo or another type of exchange, or can be used to progress the game and complete objectives. The idea is that user can interact with the game completely free, with no need to "buy" tokens (from exchanges or from in game transactions), however he will ned to pay IoTeX gas fees. This is the idea for the game.

This game will bring great interaction with the raw IoTeX network via console, as I will attempt to compile a full ioctl console into the game. People will gain exposure to using commands via web console and might teach kids how to interact with advanced blockchain networks.

Best Regards,
Jeremi Rynkiewicz

**Comment by simonerom** [comment]
Ho Jeremy, would this be the same as using the ioctl command line client (https://docs.iotex.io/developer/get-started/ioctl-install.html) on a website, or it adds more to the experience compared to ioctl?

**Comment by jrynkiew** [comment]
It will add more experience than the standard ioctl, but I will attempt to implement the full command console in the web, yes.
In the future I will add scripted scenarios and objectives to complete.

I am limited mostly by the iotex api if I want to add my own functionality, if I want to integrate into the iotex network, without hosting my own server, (which I will do in the future anyway most probably)

**Comment by simonerom** [comment]
Hi @jrynkiew, thank you for explaining your idea in more detail. 

This sounds like an interesting, ideas-rich development, however it's kinda mixing different concepts together (bug bounties, api web console, 2d gaming) and it's not very clear how all these pieces would fit together: would it maybe make more sense to split it into multiple more focused projects, and start with requesting the grant for one of them? 

For example, if you have the skills to develop a complete for a 2d/2.5d blockchain game, that's a good point to start: a new 2d game with NFTs prizes and/or ERC20 as rewards would be a nice addition to IoTeX ecosystem for sure! 

Whether you want to submit a grant request for your initial idea or for a smaller part, you must break it down into milestones with associated deliverables that can be easily verified, and with the portion of the bounty that is required for each milestone. You can use this [RFP template ](https://github.com/iotexproject/halogrants/blob/master/rfp-proposals/rfp-proposal-template1.md) as a reference: it's helpful on your side to get an idea of the the time & effort, and on our side we need it to to verify the deliverables at each milestone.

Feel free to reach out to me so that I can better support you on this request! We can also jump on a call and try to figure out what goal we can set for this project!

**Comment by jrynkiew** [comment]
Thank you Simone for your input. I will re-request a grant for one of the ideas which I am currently working on, and the rest of the ideas will be kept for a later date. Ideally I will integrate all 3 ideas into the game - bug bounties, 2d game, web console, but for now you are right, I will divide up the task and integrate each of those ideas one by one. I am focusing now on the web console, so that is what I will request with the rfp proposal system. Once that is done we can continue further.
Thank you once again, Simone, thanks for your time.

**Comment by jrynkiew** [comment]
Changing scope of JRPC submission to be more focused on web console developement and JRPC XRC20 token distribution & development

**Comment by clairebiasco** [comment]
Hi there! We are officially moving all Halo Grant proposals to our new community forum page and will no longer be accepting proposals on GitHub. Please submit your proposal to the community page here according to the provided template: https://community.iotex.io/c/halo-grants/61

Thank you!
